UI/src/assetTypeConfiguration.js

```diff
@@ -1095,7 +1103,29 @@
           var functionFn = _.find(validations, function(validation){ return _.includes(validation.types, parseInt(Property.getPropertyValue('Tyyppi', selectedAsset.get())));});
           var suggestedBoxValue = !!parseInt(_.find(selectedAsset.get().propertyData, function(asset) { return asset.publicId === "suggest_box"; }).values[0].propertyValue);
           var suggestedAssetCondition = !(suggestedBoxValue && authorizationPolicy.isMunicipalityMaintainer()) || authorizationPolicy.isOperator();
-          return (functionFn ?  functionFn.validate(Property.getPropertyValue('Arvo', selectedAsset.get())) : true) && suggestedAssetCondition;
+          var isValidFunc = (functionFn ?  functionFn.validate(Property.getPropertyValue('Arvo', selectedAsset.get())) : true)  && suggestedAssetCondition;
+
+          /* Begin: Special validate for roadwork sign */
+          var fields = selectedAsset.get().propertyData;
+          var dateValueStartField = _.filter(fields, function(field) { return field.publicId === 'trafficSign_start_date'; });
+          var dateValueEndField = _.filter(fields, function(field) { return field.publicId === 'trafficSign_end_date'; });
+          var trafficSignTypeField = _.filter(fields, function(field) { return field.publicId === 'trafficSigns_type'; });
```

Comment:
I think _.find would be better than _.filter to avoid using _.head in the moments you need to get the value?

Reply:
done